### PR TITLE
Removing mpiTasksPerNode setting

### PR DESCRIPTION
**Pull request by mbk-tp**
## What is the change?

The case settings parameter `mpiTasksPerNode` is removed from ARMI. The parameter is shifted to an internal plugin.

## Why is the change being made?

`mpiTasksPerNode` is a HPC cluster related parameter and ARMI is not the appropriate place for this parameter. 

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Review by opotowsky — COMMENTED** [review]
This looks good! Hold off on merging though. We need to time this with other updates

**Comment by john-science** [comment]
Can you add a release not here: https://github.com/terrapower/armi/blob/main/doc/release/0.4.rst?plain=1#L42

It should says "Removing setting mpiTasksPerNode."

File: doc/release/0.4.rst
Comment (john-science) [inline doc/release/0.4.rst:43]:
```suggestion
#. Removing setting ``mpiTasksPerNode``. (`PR#1953 <https://github.com/terrapower/armi/pull/1953>`_)
```

**Review by john-science — APPROVED** [review]
Approved, pending

**Comment by mbk-tp** [comment]
Using #1958

**Pull request closed without merging**
